CMakeLists.txt

```
@@ -569,19 +570,27 @@ if(BUILD_GRAPHBOLT)
     add_custom_target(
       graphbolt
       ALL
-      ${CMAKE_COMMAND} -E env
+      ${CMAKE_COMMAND} -E env ROCM_PATH=${ROCM_PATH} CC=${CC} CXX=${CXX}
+      GENERATOR=${CMAKE_GENERATOR}
       CMAKE_COMMAND=${CMAKE_CMD}
+      CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
+      CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      BINDIR=${BINDIR}
+      SRCDIR=${SRCDIR}
       CUDA_TOOLKIT_ROOT_DIR=${CUDA_TOOLKIT_ROOT_DIR}
       USE_CUDA=${USE_CUDA}
       USE_ROCM=${USE_ROCM}
       USE_LIBURING=${USE_LIBURING}
-      BINDIR=${CMAKE_CURRENT_BINARY_DIR}
+      CMAKE_HIP_ARCHITECTURES=${CMAKE_HIP_ARCHITECTURES}
```

I just tried this out and I think this needs to be escaped like `CUDAARCHS` below. Otherwise it gets screwed up if you've got multiple.

```suggestion
      CMAKE_HIP_ARCHITECTURES="${HIP_ARCHITECTURES_ESCAPED}"
```